### PR TITLE
Add logic to delete item from list

### LINKED ISSUE #11: 11. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.
It’s possible our users may make an error when entering an item or may decide they won’t be buying a certain item again in the future. Let’s help them keep their lists tidy!

AC:

* User is able to delete an item from the shopping list
* Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
* Deletion should cause the associated record(s) in the database to be deleted

Notes:

* As noted in the wireframe https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png activating the "Delete" link should pop-up a confirmation dialog because this is a "destructive" action.
* Many teams jump to a modal as the solution here, but modals are fraught with accessibility concerns, so consider using a simple JavaScript confirm dialog.

### PR DESCRIPTION
**Pull request by HonduranCoder**
## Description

Added Delete to List Items

## Related Issue
Closes #11 

## Acceptance Criteria
Screen Shot 2022-05-10 at 6 28 52 PM

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

No Delete Option

### After

Screen Shot 2022-05-10 at 6 29 25 PM

Screen Shot 2022-05-10 at 6 29 53 PM

## Testing Steps / QA Criteria

Checkout branch, pull.

### CODE REVIEW

**Comment by jmc617** [comment]
The code was concise with an easy-to-understand flow. Code also functioned as expected. Great work!

File: src/pages/ItemList.js [hunk 1]
```diff
@@ -124,6 +124,17 @@ function ItemList({ token }) {
     setSearchInput(e.target.value);
   };
 
+  const handleDelete = async (id, item_name) => {
+    const confirmation = window.confirm(
+      `Are you sure you want to delete ${item_name}?`,
```
Comment (honzlavender) [inline src/pages/ItemList.js:139]: I of course know about alert() but wasn't familiar with confirm() and how you can use it as a condition! This is v cool

**Review by jmc617 — APPROVED** [review]
The code was concise with an easy-to-understand flow. Code also functioned as expected. Great work!

File: src/pages/ItemList.js (hunk 1 above)
Comment (MikeBLambert) [reply src/pages/ItemList.js:139]: Agree that this is super cool!

**Review by MikeBLambert — APPROVED** [review]
Beautifully simple solution! Great work!

**Pull request merged**